src/backends/torch/torchdataaug.cc

```diff
@@ -279,11 +428,115 @@ namespace dd
     // get perpective matrix
 #pragma omp critical
     {
-      getQuads(src.rows, src.cols, inputQuad, outputQuad);
+      getQuads(src.rows, src.cols, cp, inputQuad, outputQuad);
     }
 
     // warp perspective
     cv::Mat lambda = cv::getPerspectiveTransform(inputQuad, outputQuad);
     cv::warpPerspective(src_enlarged, src, lambda, src.size());
+
+    if (store_rparams)
+      cp._lambda = lambda;
+  }
+
+  void TorchImgRandAugCV::warpBBoxes(std::vector<std::vector<float>> &bboxes,
+                                     cv::Mat lambda)
+  {
+    std::vector<std::vector<float>> nbboxes;
+    for (size_t i = 0; i < bboxes.size(); ++i)
+      {
+        std::vector<float> bbox = bboxes.at(i);
+        std::vector<cv::Point2f> origBBox;
+        std::vector<cv::Point2f> warpedBBox;
+
+        cv::Point2f p1;
+        p1.x = bbox[0]; // xmin
+        p1.y = bbox[1]; // ymin
+        origBBox.push_back(p1);
+        cv::Point2f p2;
+        p2.x = bbox[2]; // xmax
+        p2.y = bbox[3]; // ymax
+        origBBox.push_back(p2);
+        cv::Point2f p3;
+        p3.x = bbox[0]; // xmin
+        p3.y = bbox[3]; // ymax
+        origBBox.push_back(p3);
+        cv::Point2f p4;
+        p4.x = bbox[2]; // xmax
+        p4.y = bbox[1]; // ymin
+        origBBox.push_back(p4);
+
+        cv::perspectiveTransform(origBBox, warpedBBox, lambda);
+        float xmin = warpedBBox[0].x;
+        float ymin = warpedBBox[0].y;
+        float xmax = warpedBBox[0].x;
+        float ymax = warpedBBox[0].y;
+        for (int i = 1; i < 4; ++i)
+          {
+            if (warpedBBox[i].x < xmin)
+              xmin = warpedBBox[i].x;
+            if (warpedBBox[i].x > xmax)
+              xmax = warpedBBox[i].x;
+            if (warpedBBox[i].y < ymin)
+              ymin = warpedBBox[i].y;
+            if (warpedBBox[i].y > ymax)
+              ymax = warpedBBox[i].y;
+          }
+
+        std::vector<float> nbox = { xmin, ymin, xmax, ymax };
+        nbboxes.push_back(nbox);
+      }
+    bboxes = nbboxes;
+  }
+
+  void TorchImgRandAugCV::filterBBoxes(std::vector<std::vector<float>> &bboxes,
+                                       const int &img_width,
+                                       const int &img_height)
+  {
+    std::vector<std::vector<float>> nbboxes;
+    for (size_t i = 0; i < bboxes.size(); ++i)
+      {
+        std::vector<float> bbox = bboxes.at(i);
+        if (bbox[2] >= 0.0 && bbox[0] <= img_width && bbox[3] >= 0.0
+            && bbox[1] <= img_height)
+          {
+            std::vector<float> nbbox;
+            nbbox.push_back(std::max(0.0f, bbox[0])); // xmin
+            nbbox.push_back(std::max(0.0f, bbox[1])); // ymin
+            nbbox.push_back(
+                std::min(static_cast<float>(img_width), bbox[2])); // xmax
+            nbbox.push_back(
+                std::min(static_cast<float>(img_height), bbox[3])); // ymax
+            float surfbb = (bbox[2] - bbox[0]) * (bbox[3] - bbox[1]);
```

Comment:
There is a function `area` in [utils/bbox.hpp](https://github.com/jolibrain/deepdetect/blob/master/src/utils/bbox.hpp)

Reply:
OK, not necessary at this stage I believe, as it should be rewritten/templated for float. Overkill, no ?